File: app/Models/TouristAttraction.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;

class TouristAttraction extends Model
{
    use HasFactory;

    protected $fillable = [
        'name',
        'description',
        'address',
        'latitude',
        'longitude',
        'category',
        'images',
        'avg_rating',
        'total_reviews',
    ];

    protected $casts = [
        'images' => 'array',
        'latitude' => 'float',
        'longitude' => 'float',
        'avg_rating' => 'float',
    ];

    public function reviews(): HasMany
    {
        return $this->hasMany(Review::class);
    }

    public function itineraryItems(): HasMany
    {
        return $this->hasMany(ItineraryItem::class);
    }

    public function updateRating(): void
    {
        $reviews = $this->reviews;
        $totalReviews = $reviews->count();
        
        if ($totalReviews > 0) {
            $avgRating = $reviews->avg('rating');
            $this->update([
                'avg_rating' => $avgRating,
                'total_reviews' => $totalReviews,
            ]);
        }
    }
}

File: lib/models/tourist_attraction.dart
class TouristAttraction {
  final int id;
  final String name;
  final String description;
  final String address;
  final double latitude;
  final double longitude;
  final String category;
  final List<String> images;
  final double avgRating;
  final int totalReviews;

  TouristAttraction({
    required this.id,
    required this.name,
    required this.description,
    required this.address,
    required this.latitude,
    required this.longitude,
    required this.category,
    required this.images,
    required this.avgRating,
    required this.totalReviews,
  });

  factory TouristAttraction.fromJson(Map<String, dynamic> json) {
    return TouristAttraction(
      id: json['id'],
      name: json['name'],
      description: json['description'],
      address: json['address'],
      latitude: json['latitude'].toDouble(),
      longitude: json['longitude'].toDouble(),
      category: json['category'],
      images: List<String>.from(json['images'] ?? []),
      avgRating: json['avg_rating'].toDouble(),
      totalReviews: json['total_reviews'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'description': description,
      'address': address,
      'latitude': latitude,
      'longitude': longitude,
      'category': category,
      'images': images,
      'avg_rating': avgRating,
      'total_reviews': totalReviews,
    };
  }
}
